### ISSUE TITLE
In robust mode a ray can fail to hit a triangle mesh.

### ISSUE THREAD

**Issue by ccoustet** [embree/embree#181]
Not sure this is the same problem than reported in issue #174.
1) From a scene in which a given ray hits the expected mesh (exact hit is at a vertex)
   (10 nested cylinders in a Russian-doll-like setting, the ray goes from cylinder #5 to cylinder #6)
2) Add another mesh, not in the region crossed by the ray
   (1 more external cylinder)
3) The ray now fails to hit the previously hit mesh (hits cylinder #8, missing #6 and #7)
4) Windows only

Tested with both 2.17.1 and 2.17.3, Windows 10 x64.
Included a small project.
[cylinder.zip](https://github.com/embree/embree/files/1818611/cylinder.zip)

**Comment by svenwoop** [comment]
The ray shoots exactly at a vertex. In these cases our robust intersector may shoot through the geometry. What can help is adding an 4*float(ulp) epsilon in triangle_intersector_pluecker.h when comparing minUVW >= 0.0f and maxUVW <= 0.0f, but that is a hack and does not work in all situations either.

**Comment by svenwoop** [comment]
BTW, the CHECK(rtcDeleteDevice(device)) line fails as the device handle used in the CHECK makro is invalid after rtcDeleteDevice. Use nullptr in rtcGetDeviceError to get errors from device destruction.

**Comment by loopless** [comment]
I would agree with svenwoop. Even adding the epsilon hack does not fix the issue in all cases.  I still get rays that "miss" even though the geometry is watertight.
